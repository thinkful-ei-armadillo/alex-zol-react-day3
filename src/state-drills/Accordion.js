import React from 'react';

class Accordion extends React.Component {
    static defaultProps = {
        sections: [],
    };

    state = {
        activeSection: null,
    };

    handleOnButtonClick = (index) => {
        this.setState({ activeSection: index });
    };

    getSections = () => {
        return this.props.sections.map((section, index) => (
            <li key={index}>
                <button onClick={() => this.handleOnButtonClick(index)}>{section.title}</button>
                {(index === this.state.activeSection) && <p>{section.content}</p>}
            </li>
        ));
    };

    render() {
        return (
            <ul>{this.getSections()}</ul>
        );
    }
}

export default Accordion;