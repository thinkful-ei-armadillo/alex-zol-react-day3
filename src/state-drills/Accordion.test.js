import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordion from './Accordion';

const sectionProps = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ]

describe(`Tabs Component`, () => {

    it('renders empty when sections not supplied', () => {
        const wrapper = shallow(<Accordion />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders no sections active by default', () => {
        const wrapper =shallow(<Accordion sections={sectionProps}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders a clicked section', () => {
        const wrapper = shallow(<Accordion sections={sectionProps}/>)
        wrapper.find('button').at(0).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('only renders last clicked section', () => {
      const wrapper = shallow(<Accordion sections={sectionProps}/>);
      const buttons = wrapper.find('button');
      buttons.forEach(button => {
        button.simulate('click')
      })
      expect(toJson(wrapper)).toMatchSnapshot();
    })
  })