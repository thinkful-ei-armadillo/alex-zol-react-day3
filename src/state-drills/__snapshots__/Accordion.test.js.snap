// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Component only renders last clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`Tabs Component renders a clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Tabs Component renders empty when sections not supplied 1`] = `<ul />`;

exports[`Tabs Component renders no sections active by default 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
